stats/metrics.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package stats
+
+import "maps"
+
+// Metric is an identifier for a metric.
+type Metric string
```

Comment:
Don't we want all these to be in experimental for now, too?

Reply:
The stats package is entirely experimental as per top level package comment. I think it's fine to leave this here, since configures unstable OTel which lives in this package too.

Reply:
If you're going with that philosophy then why have `experimental/stats` for the other stuff?  Let's keep it together.

> since configures unstable OTel which lives in this package too

OTel should not have anything in the stats package, right?

Reply:
K will move.